openapi_core/schemas.py

```diff
@@ -167,7 +177,7 @@ def __init__(self, dereferencer):
     def create(self, schema_spec):
         schema_deref = self.dereferencer.dereference(schema_spec)
 
-        schema_type = schema_deref.get('type')
+        schema_type = schema_deref.get('type', 'object')
```

Comment:
Why you think `object` is default schema type? I couldn't find it in the doc.

Reply:
Which doc? Could you post a link?

Reply:
Ah, thanks for the explanation!

I'm not sure if I feel that a json schema paradigm is the way to look at this.

Consider test_post_pets. If SchemaFactory defaults to an object then the response body is correctly parsed as a PetCreate object. However, if the type is left blank then the body remains a string and the test fails.

I believe what is happening (you would know this better) is that the code checks the schema for PetCreate to determine how to parse the response body. In this case it is absolutely correct to consider it an object because it is.

PetCreate:
      x-model: PetCreate
      allOf:
        - $ref: "#/components/schemas/PetCreatePartOne"
        - $ref: "#/components/schemas/PetCreatePartTwo"

If we translate the PetCreate to json it becomes even more obvious:

{
  "PetCreate": {
    "x-model": "PetCreate",
    "allOf": [
      {
        "$ref": "#/components/schemas/PetCreatePartOne"
      },
      {
        "$ref": "#/components/schemas/PetCreatePartTwo"
      }
    ]
  }
}

In either case we end up with a paradigm where it is more correct to treat untyped members as object since that is the convention of the OAS 3.0 spec.

Reply:
You are right it will fail but `x-model` is an extension and not part of spec and shouldn't be considered in schema type resolution process. The problem is i can't find in OAS spec that this is `the convention of the OAS 3.0 spec` but I can't see any other solution.